### ISSUE TITLE
Enter keydown on items in SubMenu

### ISSUE THREAD

**Issue by lipedeitos** [react-component/menu#92]
Hi!

I'm wrapping some RC features like rc-menu inside my own componentes. 
This is how I'm using them right now:

```
const menu = (
      <FFMenu onClick={(item) => { console.log(item.key); }}>
        <FFMenu.Item key="1" icon="person">Testando 1</FFMenu.Item>
        <FFMenu.Item key="2" icon="build">Testando 2</FFMenu.Item>
        <FFMenu.SubMenu key="3" icon="check" title="Testando 3">
          <FFMenu.Item key="3-1">Testando 3-1</FFMenu.Item>
          <FFMenu.Item key="3-2">Testando 3-2</FFMenu.Item>
          <FFMenu.Item key="3-3">Testando 3-3</FFMenu.Item>
          <FFMenu.Item key="3-4">Testando 3-4</FFMenu.Item>
        </FFMenu.SubMenu>
        <FFMenu.Item key="4" icon="credit_card">Testando 4</FFMenu.Item>
      </FFMenu>
);
```

So, as you can see in the component FFMenu i pass the "onClick" function.
When i select the first item "Testando 1" and use arrows it'll navigate normally, when i press "Enter" the console.log(); shows up.
The problem is that when I'm using the arrows in any Item inside a SubMenu when i press enter, it'll close the SubMenu, and will not select the Item that was "hovered/active".

Checked the demos and examples and seems to be the same thing.
When we press enter in any item, the standard behavior should not be to "click" the item?

Sorry about my english and thanks for the attention.

**Comment by benjycui** [comment]
Yes, there are so many bugs while using keyboard, and we have no time to fix those bugs now. But PR is welcomed, thanks.

**Comment by mr2day** [comment]
Hello,

Please forgive me that I ask here, but how can I make all other SubMenus close, when I click on a MenuItem?

Thanks!

**Comment by benjycui** [comment]
@mr2day https://ant.design/components/menu/#components-menu-demo-sider-current

**Comment by mr2day** [comment]
Thanks a million! I didn't know about antd, will use all library in my project.

**Comment by rajatkantinandi** [comment]
Facing the same issue. Can't click links inside the submenu with keyboard..
Need help..

**Comment by nrps9909** [comment]
I rechecked this against current `react-component/menu` master at exact head `5e2a3e21a39af9ea4a1ec33926f1e1ffb11474a4`.

The reported interaction now works when the keyboard event is dispatched from the focused menu item, as it is in a browser:

1. focus the inline SubMenu title with ArrowDown
2. open it with Enter
3. move to its child MenuItem with ArrowDown
4. press Enter on that focused child

A temporary focused regression asserted that the root `onClick` fires exactly once with `key === "child"`. The complete `tests/Keyboard.spec.tsx` scope passed 11/11 tests and its snapshot on the exact head. I removed the temporary probe afterward, leaving the worktree clean.

The earlier apparent failure from firing the final Enter directly on the root `<ul>` was not a valid browser reproduction: it bypassed the focused MenuItem's own keydown handler. Based on the focused-element result, the original bug appears resolved in the current implementation and this issue should be eligible for maintainer closure.

AI assistance disclosure: Codex was used to inspect the current keyboard path, construct and run the temporary regression, and draft this status update. I verified the exact head, event target, callback payload, and test result locally.
